lisa/tools/sockperf.py

```diff
@@ -0,0 +1,162 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+import re
+from decimal import Decimal
+from typing import TYPE_CHECKING, Any, Dict, List, cast
+
+from lisa import notifier
+from lisa.executable import Tool
+from lisa.messages import NetworkLatencyPerformanceMessage, create_perf_message
+from lisa.operating_system import BSD, Posix
+from lisa.tools import Gcc, Git, Make
+from lisa.util import constants
+from lisa.util.process import Process
+
+if TYPE_CHECKING:
+    from lisa.testsuite import TestResult
+
+SOCKPERF_TCP = "tcp"
+SOCKPERF_UDP = "udp"
+
+
+class Sockperf(Tool):
+    @property
+    def command(self) -> str:
+        return "sockperf"
+
+    @property
+    def can_install(self) -> bool:
+        # FIXME: bug in BSD bulld is blocking BSD runs.
+        # Fast fail if attempting ot install.
+        return self.node.is_posix
+
+    _sockperf_repo = "https://github.com/Mellanox/sockperf.git"
+
+    sockperf_result_regex = re.compile(
+        r"sockperf: ---> <MAX> observation =\s+(?P<max_latency_us>[0-9]+\.[0-9]+)\s+\
+sockperf: ---> percentile 99\.999 =\s+(?P<latency99_999_percentile_us>.[0-9]+\.[0-9]+)\s+\
+sockperf: ---> percentile 99\.990 =\s+(?P<latency99_990_percentile_us>.[0-9]+\.[0-9]+)\s+\
+sockperf: ---> percentile 99\.900 =\s+(?P<latency99_900_percentile_us>.[0-9]+\.[0-9]+)\s+\
+sockperf: ---> percentile 99\.000 =\s+(?P<latency99_percentile_us>[0-9]+\.[0-9]+)\s+\
+sockperf: ---> percentile 90\.000 =\s+(?P<latency_us_90>[0-9]+\.[0-9]+)\s+\
+sockperf: ---> percentile 75\.000 =\s+(?P<latency_us_75>[0-9]+\.[0-9]+)\s+\
+sockperf: ---> percentile 50\.000 =\s+(?P<latency_us_50>[0-9]+\.[0-9]+)\s+\
+sockperf: ---> percentile 25\.000 =\s+(?P<latency_us_25>[0-9]+\.[0-9]+)\s+\
+sockperf: ---> <MIN> observation =\s+(?P<min_latency_us>[0-9]+\.[0-9]+)",
+    )
+
+    def _get_protocol_flag(self, mode: str) -> str:
+        assert (
```

Comment:
use assert_that like `in`, so if it's unexpected value, it will be printed out in the log line.